nilearn/image/image.py

```diff
@@ -1611,11 +1611,16 @@ def concat_imgs(
     auto_resample=False,
     verbose=0,
 ):
-    """Concatenate a list of 3D/4D niimgs of varying lengths.
+    """Concatenate a list of images of varying lengths.
+
+    The image list can contain:
+
+    - Niimg-like objects of varying dimensions (i.e., 3D or 4D)
+      as well as different 3D shapes and affines,
+      as they will be matched to the first image in the list
+      if ``auto_resample=True``.
 
-    The niimgs list can contain niftis/paths to images of varying dimensions
-    (i.e., 3D or 4D) as well as different 3D shapes and affines, as they
-    will be matched to the first image in the list if auto_resample=True.
+    - surface images of varying dimensions (i.e., 1D or 2D)
```

Comment:
Should we make clear that the images have to be associated with the same mesh ?

Reply:
yeah good point because that's implicit at the moment (well it will be made very explicit when users get an error)

Reply:
actually we should run 'check_same_n_vertices' more systematically on surface image operation in this module

also @man-shu, shouldn't we more often check that meshes are the same rather than just check they have same number of vertices?

we check that polymesh have the same parts and same number of vertices but 2 meshes can have same number of vertices and be wildly different: I think this would not be caught at the moment.

we have `assert_polymesh_equal` in nilearn.surface._testing

Reply:
ok quick check and pretty sure that every call of check_same_n_vertices should be replaced by assert_polymesh_equal.

Reply:
outside of the scope of this PR though